### ISSUE TITLE
Dispatching events outside of a command

### ISSUE THREAD

**Issue by cmodijk** [SimpleBus/SymfonyBridge#28]
We use SimpleBus inside a legacy application which does not use commands an handlers for everything but we do have events inside of our doctrine entity's and if i trigger the events inside a entity outside a command handler the events are not dispatched. Is there a way to manually dispatch the events inside of the event recorder queue?

**Comment by cmodijk** [comment]
I just figured out a solution https://gist.github.com/cmodijk/b5747fd2b05868600231 i'm now using this service to manually dispatch the events after the `$em->flush()` of doctrine.

**Comment by matthiasnoback** [comment]
@cmodijk that's the right solution - in fact - don't you think it would be nice to have this class in SimpleBus/MessageBus as a separate responsibility? This is mainly a copy/paste from the `HandlesRecordedMessagesMiddleware`, right?

Only thing is: I'm also rethinking exception handling (see https://github.com/SimpleBus/MessageBus/issues/38), so this part of the code might be changed anyway.

**Comment by cmodijk** [comment]
@matthiasnoback 

> don't you think it would be nice to have this class in SimpleBus/MessageBus as a separate responsibility?

Yes i think it would be nice to add this to `SimpleBus/MessageBus` but i was waiting for your response to create a pull request. I will create a pull request this weekend. 

> This is mainly a copy/paste from the HandlesRecordedMessagesMiddleware, right?

Yes!

> Only thing is: I'm also rethinking exception handling (see SimpleBus/MessageBus#38), so this part of the code might be changed anyway.

I will read that one.

**Comment by cmodijk** [comment]
I just created the pull requests for the above idea but i was thinking that it would be nice to dispatch the events automatically if doctrine `$em->flush()` is used outside a command handler / transaction. We could use the [`postFlush`](http://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/reference/events.html#lifecycle-events) doctrine event to listen if the flush is finished but we need a way to detect if we are using the commands any ideas on this?

**Comment by matthiasnoback** [comment]
Thanks so far! I'll think about this and get back to you later!

**Comment by cmodijk** [comment]
@matthiasnoback No problem we still need to upgrade our project to the latest SimpleBus versie to be able to use my pull requests.

**Comment by matthiasnoback** [comment]
@cmodijk I hope you don't mind, but I'll put this on hold for now. I'm not decided yet how and if this should evolve as part of the existing codebase.

**Comment by cmodijk** [comment]
@matthiasnoback No problem!

**Comment by krizon** [comment]
Any further thoughts on this? Looking for a similar solution in a similar use case.

**Comment by cmodijk** [comment]
@krizon I was thinking to create a separate set of packages for 'legacy' apps but i'm not sure what @matthiasnoback idea's are on this.

**Comment by yourwebmaker** [comment]
Same problem here. I want to use the SimplesBus inside a Legacy Symfony application that contains contains some entities and some domain events. 

The entities already record the Domain events through `ContainsRecordedMessages` trait, however, I'm not able to (re-)dispatch such events to some Subscribers (EmailSubscriber, AsanaSubscriber, etc.).

Here is my service definition:

```yaml
    foo.test_subscriber:
        class: Foo\TestSubscriber
        tags:
            - { name: event_subscriber, subscribes_to: Foo\SomethineAdded }
        arguments:
            - '@command_bus'
```
@cmodijk @krizon Did you solve this problem?

Thanks.

**Comment by cmodijk** [comment]
@yourwebmaker Tnx for reaching out. Wel this is a very old topic and I created these pull request a long time ago to solve the problem https://github.com/SimpleBus/MessageBus/pull/42. In the mean time I took over the project but these things are not merged yet. The reason for that is that I believe know that we should create a separate package to support the Legacy apps. But i did not find the time and also interest of others to make this happen yet. You should be able to take the classes from my pull request and add these to your own project.

**Comment by yourwebmaker** [comment]
@cmodijk thanks for replying me back. 

I also tried to use your solution but no success so far.

We are trying reuse the `PrivateMessageRecorderCapabilities` functionality provided by the SimpleBus, so we collect the Domain events recorded in our entities. After flushing the entities the events are collected. This is working perfectly. 👍 

**So, what we want to do with those events?** We want to handle them.

**How do we want to handle those events?** Since we already have some subscribers already in place, we do want to reuse them. They are actually `Symfony` customized subscribers (like emailsubscriber, asanasubcriber). I want to handle all those events inside such subscribers. Do you know if is it possible using `SimpleBus`?

Thanks.
